### PR TITLE
SAAS-2468: Google Compute Engine Instance Maintenance Behavior

### PR DESCRIPTION
**Pull request by mehakseedat63**

### CODE REVIEW

File: plugins/google/compute/instanceMaintenanceBehavior.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+var async = require('async');
+var helpers = require('../../../helpers/google');
+
+module.exports = {
+    title: 'Instance Maintenance Behavior',
+    category: 'Compute',
+    description: 'Ensure that On Host Maintenance configuration is set to Migrate for VM instances.',
```
Comment (AkhtarAmir) [inline plugins/google/compute/instanceMaintenanceBehavior.js:7]:
```suggestion
    description: 'Ensure that "On Host Maintenance" configuration is set to Migrate for VM instances.',
```

File: plugins/google/compute/instanceMaintenanceBehavior.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    more_info: "When Google Compute Engine performs regular maintenance of its infrastructure it live migrate your VM instances to other hardware if you configured the instance's availability policy to use live migration. This prevents your applications from experiencing disruptions during these events.",
```
Comment (AkhtarAmir) [inline plugins/google/compute/instanceMaintenanceBehavior.js:8]:
```suggestion
    more_info: "When Google Compute Engine performs regular maintenance of its infrastructure, it migrates your VM instances to other hardware if you have configured the instance's availability policy to use live migration. This prevents your applications from experiencing disruptions during these events.",
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

**Comment by AkhtarAmir** [comment]
Can you fix linting issues?

**Review by AkhtarAmir — APPROVED** [review]

File: plugins/google/compute/instanceMaintenanceBehavior.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+                if (!instances.data.length) {
+                    noInstances.push(zone);
+                    return zcb();
+                }
+
+                instances.data.forEach(instance => {
+                    let resource = helpers.createResourceName('instances', instance.name, project, 'zone', zone);
+                   
+                    if (instance.scheduling && instance.scheduling.onHostMaintenance && instance.scheduling.onHostMaintenance == 'MIGRATE') {
```
Comment (AkhtarAmir) [inline plugins/google/compute/instanceMaintenanceBehavior.js:51]:
```suggestion
                    if (instance.scheduling && instance.scheduling.onHostMaintenance && instance.scheduling.onHostMaintenance.toUpperCase() == 'MIGRATE') {
```

**Review by dipsubha06 — APPROVED** [review]

**Pull request merged**